example/maker_pi_pico.hw_config.c

```diff
@@ -0,0 +1,98 @@
+/* hw_config.c
+Copyright 2021 Carl John Kugler III
+
+Licensed under the Apache License, Version 2.0 (the License); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an AS IS BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+/*
+
+This file should be tailored to match the hardware design.
+
+There should be one element of the spi[] array for each hardware SPI used.
+
+There should be one element of the sd_cards[] array for each SD card slot.
+The name is should correspond to the FatFs "logical drive" identifier.
+(See http://elm-chan.org/fsw/ff/doc/filename.html#vol)
+The rest of the constants will depend on the type of
+socket, which SPI it is driven by, and how it is wired.
+
+*/
+
+#include <string.h>
+//
+#include "my_debug.h"
+//
+#include "hw_config.h"
+//
+#include "ff.h" /* Obtains integer types */
+//
+#include "diskio.h" /* Declarations of disk functions */
+
+void spi0_dma_isr();
+
+// Hardware Configuration of SPI "objects"
+// Note: multiple SD cards can be driven by one SPI if they use different slave
+// selects.
+static spi_t spis[] = {  // One for each SPI.
+    {
+        .hw_inst = spi1,  // SPI component
+        .miso_gpio = 12, // GPIO number (not pin number)
+        .mosi_gpio = 11,
+        .sck_gpio = 10,
+        /* The choice of SD card matters! SanDisk runs at the highest speed. PNY
+           can only mangage 5 MHz. Those are all I've tried. */
+        //.baud_rate = 1000 * 1000,
+        .baud_rate = 12500 * 1000,  // The limitation here is SPI slew rate.
+        //.baud_rate = 6250 * 1000,  // The limitation here is SPI slew rate.
+        //.baud_rate = 25 * 1000 * 1000, // Actual frequency: 20833333. Has
+        // worked for me with SanDisk.
+
+        // Following attributes are dynamically assigned
+        .dma_isr = spi0_dma_isr,
+        .initialized = false,  // initialized flag
+    }};
+
+// Hardware Configuration of the SD Card "objects"
+static sd_card_t sd_cards[] = {  // One for each SD card
+    {
+        .pcName = "0:",           // Name used to mount device
+        .spi = &spis[0],          // Pointer to the SPI driving this card
+        .ss_gpio = 15,            // The SPI slave select GPIO for this SD card
+        .card_detect_gpio = 22,   // Card detect
+        .card_detected_true = -1,  // What the GPIO read returns when a card is
```

Comment:
Since pull request #22, there is a new field:` bool use_card_detect;`, which, since it is static in this instantiation, defaults to `0` (or `false`). So, you could simplify this by getting rid these lines:
`        .card_detect_gpio = 22,   // Card detect`
`        .card_detected_true = -1,  // What the GPIO read returns when a card is`